### PR TITLE
feat(combobox): Allow for greater nesting of calcite-combobox-items within calcite-combobox

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #2337

## Summary

feat(combobox); Allow for greater nesting of calcite-combobox-items within calcite-combobox #2337

### CODE REVIEW

File: src/components/calcite-combobox/utils.ts [hunk 1]
```diff
@@ -24,12 +25,17 @@ export function hasActiveChildren(node: HTMLCalciteComboboxItemElement): boolean
...
-  if (!parent) {
+  if (!Build.isBrowser) {
     return 0;
   }
-  if (!grandparent) {
-    return 1;
-  }
-  return 2;
+
+  const result = document.evaluate(
```
Comment (paulcpederson) [inline src/components/calcite-combobox/utils.ts:32]: wow this is an advanced maneuver 🕶️ ⚡

**Review by paulcpederson — APPROVED** [review]

File: src/components/calcite-combobox/utils.ts (hunk 1 above)
Comment (driskull) [reply src/components/calcite-combobox/utils.ts:32]: Is xpath use ok @jcfranco ?

Comment (jcfranco) [reply src/components/calcite-combobox/utils.ts:32]: TBH, I need a refresher on it. The last time I dealt w/ xpath was in the good ol' Flex/ActionScript days. 😂  

How does it deal in terms of performance and does it work across shadow DOM boundaries?

Comment (driskull) [reply src/components/calcite-combobox/utils.ts:32]: I'm not sure on the performance but it seems to work fine.

It shouldn't need to work across shadow for this use case but I think it does.

**Pull request merged**